app/src/main/java/com/example/contactlistapp/view/ContactProfileAdapter.kt

```diff
@@ -0,0 +1,189 @@
+package com.example.contactlistapp.view
+
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import android.widget.PopupMenu
+import androidx.appcompat.app.AlertDialog
+import androidx.recyclerview.widget.RecyclerView
+import com.example.contactlistapp.Data.ContactProfileData
+import com.example.contactlistapp.R
+import com.example.contactlistapp.databinding.ListItemBinding
+
+
+class ContactProfileAdapter(
+    var contactProfileData: ArrayList<ContactProfileData>,
+    private val viewModel: ContactProfileViewModel
+) :
+    RecyclerView.Adapter<ContactProfileAdapter.ProfileViewHolder>() {
+
+
+    fun setFilteredList(contactProfileData: ArrayList<ContactProfileData>) {
+        this.contactProfileData = contactProfileData
+        notifyDataSetChanged()
+    }
+
+    // Store the position of the currently expanded item
+    private var expandedPosition: Int = RecyclerView.NO_POSITION
+
+    inner class ProfileViewHolder(val binding: ListItemBinding) :
+        RecyclerView.ViewHolder(binding.root)
+
+
+    // Handles the popup menu when the more options icon is clicked
+    private fun menuPopUp(view: View, position: Int) {
+
+
+        val popUpMenus = PopupMenu(view.context, view)
+        popUpMenus.inflate(R.menu.menu_edit_delete)
+        popUpMenus.setOnMenuItemClickListener { menuItem ->
+            when (menuItem.itemId) {
+                R.id.editText -> {
+                    // Handle the Edit option for the specific item at 'position'
+                    // Show an AlertDialog for editing
+                    showEditDialog(view, position)
+                    true
+                }
+
+
+                R.id.delete -> {
+                    showDeleteConfirmationDialog(view, position)
+
+                    true
+                }
+
+
+                else -> false
+
+            }
+        }
+        // Show the PopupMenu
+        val popup = PopupMenu::class.java.getDeclaredField("mPopup")
+        popup.isAccessible = true
+        val menu = popup.get(popUpMenus)
+        menu.javaClass.getDeclaredMethod("setForceShowIcon", Boolean::class.java)
+            .invoke(menu, true)
+        popUpMenus.show()
+
+
+    }
+    // Displays an edit dialog when the "Edit" option is clicked in the popup menu
+    private fun showEditDialog(view: View, position: Int) {
+        val inflater = LayoutInflater.from(view.context)
+        val editView = inflater.inflate(R.layout.addnewcontact, null)
+        val nameEditText = editView.findViewById<EditText>(R.id.etName)
+        val numberEditText = editView.findViewById<EditText>(R.id.etNumber)
```

Comment:
You can get rid of multiple findViewById() calls within the adapter by using data binding for adapter as well